File: src/main/java/pl/tsimur/jpatest/service/OrganizationServiceImpl.java
package pl.tsimur.jpatest.service;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import pl.tsimur.jpatest.model.Organization;
import pl.tsimur.jpatest.repository.OrganizationRepository;

import java.util.List;

@Slf4j
@Service
@RequiredArgsConstructor
public class OrganizationServiceImpl implements OrganizationService {

    private final OrganizationRepository organizationRepository;

    @Override
    public Integer save(Organization org) {
        Organization organization = organizationRepository.save(org);
        log.info("Saved organization with name: {}", organization.getName());
        return organization.getNip();
    }

    @Override
    public Organization find(Integer nip) {
        Organization organization = organizationRepository.findOrganizationByNip(nip);
        try {
            log.info("Found organization with name: {}", organization.getName());
        } catch (NullPointerException ex) {
            ex.getMessage();
        }
        return organization;
    }

    @Override
    public List<Organization> findAllByName(String name) {
        List<Organization> orgWithName = organizationRepository.findAllOrganizationByName(name);
        log.info("Found {} organization with name: {}", orgWithName.size(), name);
        return orgWithName;
    }

    @Override
    public List<Organization> findAllByAddress(String address) {
        List<Organization> orgWithAddresses = organizationRepository.findAllOrganizationByAddress(address);
        log.info("Found {} organizations with {} addresses", orgWithAddresses.size(), address);
        return orgWithAddresses;
    }

    /**
     * Имена компаний, где работают пенсионеры
     */
    @Override
    public List<String> findAllOrgNamesWithPensioners() {
        List<String> orgNames = organizationRepository.findAllOrgNamesWithPensioners();
        log.info("👴🏼 Found {} organizations where pensioners work", orgNames.size());
        return orgNames;
    }

    @Override
    @Transactional
    public void updateAddress(Integer nip, String address) {
        organizationRepository.updateOrganizationByNip(nip, address);
        log.info("Updated organization {} ", address);
    }

    @Override
    @Transactional
    public void delete(Integer nip) {
        organizationRepository.deleteByNip(nip);
        log.info("Deleted organization {} ", nip);
    }

}

File: src/main/kotlin/pl/timur/jpatest/service/OrganizationServiceImpl.kt
package pl.timur.jpatest.service

import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import pl.timur.jpatest.model.Organization
import pl.timur.jpatest.repository.OrganizationRepository

@Service
class OrganizationServiceImpl(
    private val organizationRepository: OrganizationRepository
) : OrganizationService {

    companion object {
        private val log = LoggerFactory.getLogger(this::class.java)
    }


    override fun save(org: Organization): Int {
        val organization = organizationRepository.save(org)
        log.info("Saved organization with name: ${organization.name}")
        return organization.nip
    }

    override fun find(nip: Int): Organization? {
        val organization = organizationRepository.findOrganizationByNip(nip)
        log.info("Saved organization with nip:{}", nip)
        return organization
    }

    override fun findAllByName(name: String): List<Organization> {
        val orgWithName = organizationRepository.findAllOrganizationByName(name)
        log.info("Found ${orgWithName.size} organization with name: $name")
        return orgWithName
    }

    override fun findAllByAddress(address: String): List<Organization> {
        val orgWithAddresses = organizationRepository.findAllOrganizationByAddress(address)
        log.info("Found ${orgWithAddresses.size} organizations with $address addresses")
        return orgWithAddresses
    }

    override fun findAllOrgNamesWithPensioners(): List<String> {
        val orgNames = organizationRepository.findAllOrgNamesWithPensioners()
        log.info("👴🏼 Found ${orgNames.size} organizations where pensioners work")
        return orgNames
    }

    @Transactional
    override fun updateAddress(nip: Int, address: String) {
        organizationRepository.updateOrganizationByNip(nip, address)
        log.info("Updated organization $address")
    }

    @Transactional
    override fun delete(nip: Int) {
        organizationRepository.deleteByNip(nip)
        log.info("Deleted organization $nip")
    }
}